test/unit_tests/autoqasm/test_operators.py

```diff
@@ -930,3 +930,52 @@ def test_list_ops():
         assert np.array_equal(c, [[2, 3, 4], [2, 3, 4]])
 
     assert test_list_ops.build().to_ir()
+
+
+class TestFloorDiv:
+    def test_integer_division_on_intvars(self):
+        @aq.main(num_qubits=2)
+        def main():
+            a = aq.IntVar(5)
+            b = aq.IntVar(2)
+            c = a // b  # noqa: F841
+
+        expected_ir = """OPENQASM 3.0;
+int[32] c;
+qubit[2] __qubits__;
+int[32] a = 5;
+int[32] b = 2;
+int[32] __int_2__;
+__int_2__ = a / b;
+c = __int_2__;"""
+        assert main.build().to_ir() == expected_ir
+
+    def test_integer_division_on_mixed_vars(self):
+        @aq.main(num_qubits=2)
+        def main():
+            a = aq.IntVar(5)
+            b = aq.FloatVar(2.3)
+            c = a // b  # noqa: F841
+
+        expected_ir = """OPENQASM 3.0;
+float[64] c;
+qubit[2] __qubits__;
+int[32] a = 5;
+float[64] b = 2.3;
+float[64] __float_2__;
+__float_2__ = a;
+float[64] __float_3__;
+__float_3__ = __float_2__ / b;
+c = __float_3__;"""
+        assert main.build().to_ir() == expected_ir
```

Comment:
For full validation of your code's correctness - you could also run this program on a simulator and validate that the results of the floor division are what you expect.

Running on a simulator is pretty simple, see here for an example: 
https://github.com/amazon-braket/autoqasm/blob/24e6fa18d04bfa984e475314a7d1fd43bbbda85d/test/unit_tests/autoqasm/test_parameters.py#L28-L33
You'd just grab `task.result().measurements` which is a dict containing the values of all of the variables in the program.

Reply:
Thank you so much, this helped a ton! Should add tests involving the local simulator to `test_operators.py` as well or was this suggestion mostly for local development? 

On the local development note, I realized that every time OpenQASM casts a FloatVar that is not a round number to an IntVar I get a warning coming from `openqasm/_helpers/casting.py:96`. For example,

```
@aq.main
def floor_div():
    a = aq.IntVar(2)
    b = 4.3
    c = b // a  # noqa: F841
```
results in the warning `UserWarning: Integer overflow for value 2.15 and size 32.`

This is not directly related to this PR since:
```
@aq.main
def int_to_float():
    a = aq.IntVar(2.15)
```
nets the same warning. Do you think I should do something in the context of this PR to silence the warning when the user does integer division or should I leave it?

Reply:
ah, I see, that's a warning from the simulator here:
https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/a4d7f98cb123ae6a1092972e728d2dbb93cb27b5/src/braket/default_simulator/openqasm/_helpers/casting.py#L83-L99

It looks like that warning should be modified to account for that case - not sure yet if that change should happen in the autoqasm repo, or in the amazon-braket-default-simulator-python repo. Feel free to open an issue here to track it though, with the details you just provided! It definitely doesn't need to be fixed within this PR.

Reply:
Amazing, will do!